src/MongoDB.Driver/MongoCollectionImpl.cs

```diff
@@ -747,8 +747,10 @@ private AggregateToCollectionOperation CreateAggregateToCollectionOperation<TRes
             };
         }
 
-        private BulkMixedWriteOperation CreateBulkWriteOperation(IEnumerable<WriteModel<TDocument>> requests, BulkWriteOptions options)
+        private BulkMixedWriteOperation CreateBulkWriteOperation(IClientSessionHandle session, IEnumerable<WriteModel<TDocument>> requests, BulkWriteOptions options)
         {
+            var effectiveWriteConcern = session.IsInTransaction ? WriteConcern.Acknowledged : _settings.WriteConcern;
+
```

Comment:
This pull request is mostly about updating the JSON transaction tests, but the new tests revealed a minor bug in the driver in this method.